### PR TITLE
test: raise release-3.17.0 UT line coverage to 60.6% (+342 cases across 13 modules)

### PR DESCRIPTION
**Pull request by kyonRay**
## Summary

Raises release-3.17.0 unit-test line coverage from 54.8% to 60.6% (CI-measured: 38,002 / 62,699 lines) by adding 342 unit-test cases across 13 modules (69 new/updated test files). The tests deliberately target error paths, edge cases and validation branches rather than only happy paths, which surfaced four real defects (filed separately, see below).

## Coverage

| point | line coverage |
|---|---|
| base `53ab118` | 54.8% |
| head (CI Coverage job, `fd960b6d`) | **60.6%** (38,002 / 62,699 lines) |

## Bugs found while writing these tests

Production fixes are out of scope for this test-only PR; each is filed as an issue:

- #5353 — RLP integer decoder accepts non-canonical (leading-zero) encodings
- #5354 — Negative numeric config values silently wrap to huge `size_t`, bypassing validation
- #5355 — `BlockImpl` `setReceipt`/`setTransaction`/`transactionHash` use unchecked indices (out-of-bounds)
- #5356 — `Sha256::hasher()` returns a SHA3-256 hasher, disagreeing with `Sha256::hash()` (pinned by `HashImplHasherTest`)

## Test cases added (342 across 69 files)

<details><summary><b>bcos-boostssl</b> — 2 files, 7 cases</summary>

`ContextConfigTest.cpp`
- settersAndGettersRoundTrip
- initConfigSslType
- initConfigSmSslType

`RawWsMessageTest.cpp`
- factoryBuildsRawMessage
- accessorsRoundTrip
- respPacketFlag
- encodeAppendsPayloadDecodeRestoresIt

</details>

<details><summary><b>bcos-codec</b> — 4 files, 19 cases</summary>

`ContractABICodecOutTest.cpp`
- abiOutAddressRoundtrip
- abiOutUnsupportedTypeReturnsFalse
- abiOutStringOffsetOutOfRangeThrows

`ContractABITypeParseTest.cpp`
- resetElementaryTypesAndValid
- resetRejectsMalformedArrayTypes
- removeExtentOnScalarReturnsFalse
- dynamicClassification
- parserRejectsMalformedSignature

`RLPTest.cpp`
- stringsEncode
- uint64Encode
- uint256Encode
- vectorsEncode
- stringsDecode
- uintDecode
- uint256Decode
- vectorsDecode
- decodeRejectsMalformedInputs

`ScaleDecoderStreamTest.cpp`
- compactIntegerTruncatedThrows
- spanAndCurrentIndexAccessors

</details>

<details><summary><b>bcos-crypto</b> — 6 files, 14 cases</summary>

`ChecksumAddressExtraTest.cpp`
- checksumEIP55KnownVectors
- checksumWithChainId
- newEVMAddressDeterminismAndOverloads
- checksumFromBytesLowercaseHex
- legacyCreateAddressNonceEncoding

`Ed25519RecoverTest.cpp`
- recoverFromInputRejectsInvalid

`HashImplHasherTest.cpp`
- hasherMatchesHashForConsistentImpls
- sha256HasherInconsistentWithHash
- sha256ReportsSha3ImplType

`SM2RecoverTest.cpp`
- recoverAddressFromInputRejectsInvalid

`Secp256k1RecoverTest.cpp`
- recoverFromInputRejectsBad
- recoverAddressRejectsBadSignature

`SignatureRoundtripTest.cpp`
- ed25519SignVerifyRecoverRoundtrip
- sm2SignVerifyAndCreateKeyPair

</details>

<details><summary><b>bcos-executor</b> — 3 files, 10 cases</summary>

`DAGTest.cpp`
- linearChainConsumesInOrder
- independentVerticesAllTopLevel
- diamondDependency
- clearIsSafeAfterGenerate

`DagTransferPrecompiledTest.cpp`
- emptyUserNameRejectedByEveryMethod
- zeroAmountRejected
- transferToSelfReturnsBeforeTable
- unknownSelectorIsIgnored

`SmallBankPrecompiledTest.cpp`
- updateBalanceAndSendPayment
- validationPaths

</details>

<details><summary><b>bcos-gateway</b> — 5 files, 23 cases</summary>

`GatewayConfigSettersTest.cpp`
- settersAndGetters
- peerBlacklistConfig
- peerWhitelistConfig

`PeerBlackWhitelistTest.cpp`
- whitelistMembershipAndSize
- whitelistDisabledMatchesAll
- whitelistUpdateAndSetEnable
- blacklistMembership
- blacklistDisabledMatchesNone

`RateLimiterStatTest.cpp`
- statCountersUpdate
- statToStringWithActivity
- keyFormatters
- updateAndFlush

`RouterTableTest.cpp`
- entrySettersRoundTrip
- tableNodeIdAndEntries
- encodeDecodeRoundTrip
- eraseRemovesEntry
- factoryBuildsFromEncoded

`TopicManagerExtraTest.cpp`
- subscribeAndQueryByClient
- topicSeqIncrements
- removeTopicsSubset
- removeTopicsByClientClearsAll
- reverseLookupClientsByTopic
- queryTopicsSubByClientJson

</details>

<details><summary><b>bcos-protocol</b> — 3 files, 11 cases</summary>

`TopicItemTest.cpp`
- constructAndAccess
- orderingPlacesIntoSet
- parseSubTopicsJsonHappyPath
- parseSubTopicsJsonEmptyTopicsArray
- parseSubTopicsJsonMalformed
- parseSubTopicsJsonObjectInArrayReturnsFalse

`TransactionStatusTest.cpp`
- toStringCoversAllNamedCases
- unknownAndDefaultFallToUnknown

`TransactionSubmitResultTest.cpp`
- defaultsMatchFactoryContract
- settersAndGettersRoundTrip
- factoryProducesIndependentInstances

</details>

<details><summary><b>bcos-rpc</b> — 11 files, 62 cases</summary>

`EventSubMatcherTest.cpp`
- emptyParamsMatchAnything
- addressFilter
- topicFilter

`EventSubRequestTest.cpp`
- unsubRequestRoundTrip
- unsubFromJsonMissingIdFails
- unsubFromJsonMissingGroupFails
- unsubFromJsonMalformedFails
- subFromJsonFullParams
- subFromJsonMissingParamsFails
- subFromJsonDefaultsWhenParamFieldsAbsent

`EventSubResponseTest.cpp`
- settersAndGenerateJson
- roundTripFromJson
- fromJsonMalformed

`GatewayHandlersTest.cpp`
- getPeersErrorPathFires
- getGroupNodeInfoUsesGroupManager

`RpcHandlersTest.cpp`
- getBlockNumberReturnsPositive
- getBlockByNumberHeaderOnly
- getBlockByNumberFullAndTxHashOnly
- getBlockHashByNumberFires
- getSystemConfigByKeyKnownKey
- groupQueryHandlersFire
- nodeListHandlersFire
- totalTransactionCountFires
- pendingTxSizeFires
- callViaAsyncWait

`RpcReadHandlersTest.cpp`
- nodeListQueriesFire
- groupNodeInfoQueryFires
- filterCreationFires

`SchedulerHandlersTest.cpp`
- getCodeReturnsHexFromScheduler
- getABIReturnsFromScheduler

`Web3ConsensusTest.cpp`
- coinbaseDerivesAddressFromConsensusNodeId
- jsonRpcConsensusAndSyncStatusHandlers

`Web3EthMethodsTest.cpp`
- getBlockByNumberLatest
- getBlockByNumberFullTxs
- getBlockTransactionCountByNumber
- getTransactionByHashUnknown
- getTransactionReceiptUnknown
- malformedParamsReportError
- getBlockTransactionCountByNumberEarliestAndPending
- getTransactionByBlockNumberAndIndex
- callOnSchedulerBackedPath
- getBlockByNumberExtendedTags
- getBalanceEmptyStorageReturnsZero
- getTransactionCountEmptyStorage
- getStorageAtEmptyStorage
- uncleMethodsReturnEmpty
- blockAndPendingFiltersRegister
- uninstallUnknownFilter
- newFilterWithParams
- getFilterChangesUnknownId
- sendRawTransactionGarbageReportsError

`Web3NodeStatusTest.cpp`
- ethSyncingResolvesViaSyncService
- netPeerCountUsesSyncPeerStatus
- netVersionReadsChainIdFromLedger
- netListeningIsConstantTrue
- maxPriorityFeePerGasIsConstant
- protocolVersionReportsNotImplemented

`Web3ResponseTest.cpp`
- combineBlockResponseGenesisBlock
- combineBlockResponseNonGenesisComputesMiner
- combineBlockResponseFullTxsEmptyList
- combineTxResponseShapesTransaction
- combineReceiptResponseShapesReceipt

</details>

<details><summary><b>bcos-sdk</b> — 8 files, 42 cases</summary>

`BlockNumberInfoTest.cpp`
- settersGetters
- fromJsonValid
- fromJsonMissingFields
- fromJsonMalformed

`ConfigTest.cpp`
- sendRpcFlagRoundTrip
- loadConfigCommonAndPeersNoSsl
- loadConfigSinglePeer

`ContractABITypeCodecLiquidTest.cpp`
- liquidBaseTypeSerialize
- liquidBaseTypeDeserialize
- liquidAbstractTypeSerialize

`EventSubParamsTest.cpp`
- settersGettersAndTopics
- verifyParamsValidAndBlockRange
- verifyParamsRejectsTooManyTopicGroups
- fromJsonStringValid
- fromJsonStringMalformed
- streamOperatorDumps

`JsonGroupInfoCodecTest.cpp`
- chainNodeRoundTrip
- groupInfoRoundTripString
- groupInfoSerializeJsonValue
- groupInfoDeserializeErrors
- chainNodeDeserializeErrors

`JsonRpcRequestTest.cpp`
- jsonRpcExceptionCarriesCodeAndMessage
- defaultFieldsHaveSpecCompliantValues
- settersRoundTrip
- toJsonProducesParseableEnvelope
- fromJsonHappyPathPopulatesAllFields
- fromJsonAcceptsRequestWithoutId
- fromJsonMissingJsonrpcThrowsInvalidRequest
- fromJsonMissingMethodThrowsInvalidRequest
- fromJsonMissingParamsThrowsInvalidRequest
- fromJsonParamsNotArrayThrowsInvalidRequest
- fromJsonMalformedJsonThrowsInvalidRequest
- factoryAssignsMonotonicallyIncreasingIds

`KeyPairBuilderTest.cpp`
- genSecp256K1KeyPairFromRandom
- genSecp256K1KeyPairFromExplicitPrivateKeyIsDeterministic
- genSm2KeyPairFromRandom
- genSm2KeyPairFromExplicitPrivateKeyIsDeterministic
- genKeyPairUnsupportedAlgorithmThrows

`SignatureTest.cpp`
- signRejectsNonHsmKeyPair
- verifyRejectsNonHsmType
- recoverRejectsNonHsmType
- recoverAddressRejectsNonHsmType

</details>

<details><summary><b>bcos-sealer</b> — 1 files, 7 cases</summary>

`SealingManagerTest.cpp`
- latestSettersRoundTrip
- resetSealingClearsSealingNumber
- setOnReadyCallbackStores
- resetSealingInfoIsIdempotentForRepeatedCall
- shouldGenerateProposalRequiresSealingInfo
- notifyResetTxsFlagWithEmptyListIsNoOp
- notifyResetProposalWithEmptyMetadataIsNoOp

</details>

<details><summary><b>bcos-security</b> — 3 files, 19 cases</summary>

`BcosKmsKeyEncryptionTest.cpp`
- directConstructorAesPath
- directConstructorSmPath
- encryptContentsAlwaysThrowsNotImplemented
- encryptFileAlwaysThrowsNotImplemented
- decryptContentsOnGarbageWrapsAsEncryptedFileError
- decryptFileMissingPathWrapsAsEncryptedFileError

`BcosKmsTest.cpp`
- setIpPortStoresUrlComposition
- uniformDataKeyKeccakBranch
- uniformDataKeySmBranchQuadruplesHash
- getDataKeyEmptyCipherThrows
- getDataKeyHappyPathUsingMockClient
- getDataKeyCacheHitSkipsCallMethod
- getDataKeyServerErrorClearsCacheAndThrows
- getDataKeyTransportThrowWrapsAsKeyCenterError
- clearCacheForcesNextCallToReFetch

`UtilsTest.cpp`
- splitOnColonProducesParts
- splitEmptyReturnsEmpty
- splitWithoutColonReturnsSingle
- splitClearsExistingContent

</details>

<details><summary><b>bcos-tars-protocol</b> — 8 files, 57 cases</summary>

`BlockHeaderImplTest.cpp`
- referenceConstructorInnerAndSetInner

`BlockImplTest.cpp`
- fromTarsAndInnerAccessors
- transactionAndReceiptByIndex
- nonceListRoundTrip

`CommonConvertTest.cpp`
- ledgerConfigRoundTrip
- toTarsLedgerConfigNullReturnsEmpty
- twoPCParamsRoundTrip
- logEntryRoundTripAndTake
- gatewayInfoRoundTrip
- toTarsGatewayInfoNullReturnsEmpty

`ErrorConverterTest.cpp`
- toTarsErrorFromBcosError
- toTarsErrorFromPtrPopulatesFields
- toTarsErrorFromNullPtrYieldsZeroCode
- toBcosErrorFromZeroIsNull
- toBcosErrorFromNonZeroWrapsCodeAndMessage
- toBcosErrorFromZeroIntIsNull
- toBcosErrorFromNonZeroIntWraps
- toUniqueBcosErrorZeroIsNull
- toUniqueBcosErrorNonZeroWraps
- roundTripPreservesAllFields
- roundTripWithZeroVersionRange

`ExecutionMessageImplTest.cpp`
- factoryProducesUsableInstance
- scalarFieldsRoundTrip
- boolFlagsRoundTrip
- dataAndKeyLocksRoundTrip
- delegateCallFieldsRoundTrip
- gasPriceFieldsRoundTrip
- takeAccessorsReturnPayload
- createSaltEmptyValueAndGarbage
- logEntriesRoundTrip
- contractTableChangedRoundTrip
- transactionHashRoundTrip
- unimplementedAccessorsThrow

`TransactionFactoryImplTest.cpp`
- buildWithAllGasFields
- buildWithKeyPairSignsAndVerifies
- encodeDecodeRoundTripWithSigCheck
- decodeTransactionStandalone
- cryptoSuiteAccessorRoundTrips
- createEmptyTransactionIsTainted
- createFromInputPreservesAllFlags
- decodeRejectsNonHexQuantityV1
- decodeAcceptsEmptyQuantityV1
- decodeRejectsHashMismatch
- builderRejectsNonHexQuantityV1
- builderV0ForcesZeroGasFields

`TransactionReceiptImplTest.cpp`
- createReceiptPopulatesCoreFields
- settersRoundTrip
- logEntriesRoundTrip
- hashThrowsBeforeCompute
- takeLogEntriesFromPayload
- innerAccessorsRoundTrip
- messageRoundTrip
- sizeGasUsedAndLogIndex

`TransactionSubmitResultImplTest.cpp`
- defaultsAreZeroed
- scalarSettersRoundTrip
- hashSettersRoundTripThroughTarsBytes
- innerReflectsMutations

</details>

<details><summary><b>bcos-tool</b> — 7 files, 52 cases</summary>

`BfsFileFactoryTest.cpp`
- buildDirEntryTypeDispatch
- buildLinkV30WritesAbiAndName
- buildLinkNewerVersionSkipsEmpty
- buildAuthWritesAdminAndFixedRows
- buildDirAndContractStubs

`NodeConfigExecutorConfigTest.cpp`
- wasmWithoutSerialExecuteRejected
- wasmWithAuthCheckRejected
- emptyAuthAdminAccountRejected
- validSolidityConfigAccepted

`NodeConfigGenesisDataTest.cpp`
- modernSectionedFormat
- legacyDashJoinedFormat

`NodeConfigLoadersTest.cpp`
- txPoolConfigRejectsZero
- ledgerConfigRejectsNonPositiveConsensusCounts
- rpcConfigDefaultsAndPopulated
- web3RpcConfigDefaultsAndPopulated
- gatewayConfigDefaultsAndPopulated
- certConfigDefaultsAndPopulated
- txPoolConfigValidAndInvalid
- chainConfigValidAndInvalid
- web3ChainConfigValidAndInvalid
- securityConfigLegacyAndKmsError
- sealerConfigValidAndInvalid
- storageSecurityConfigDisabledAndError
- syncConfigValidAndInvalid
- storageConfigDefaultsAndTikv
- failOverConfigDisabledAndError
- othersConfigDefaultsAndForceSender
- consensusConfigValidAndInvalid
- executorConfigValidAndInvalid
- ledgerConfigValidAndInvalid
- serviceConfigWithoutTars
- tarsProxyConfigFromFile
- nodeServiceConfigValidAndInvalid
- checkServiceValidAndInvalid
- getServiceNameRequireAndNot
- certSettersRoundTrip
- genesisConfigFromStringFull
- securityConfigHsmCloudKmsBcosKms
- gettersAfterFullGenesisLoad

`NodeConfigTest.cpp`
- isValidPortRejectsReservedAndOutOfRange
- defaultsAreReadableWithoutLoad
- loadConfigFromStringEmptyDoesNotLoseInvariant
- loadConfigFromStringPartialDocumentDispatchesSubLoaders
- certMaterialSettersRoundTrip
- readOnlyAccessorsQueryableOnDefault

`NodeConfigValidationTest.cpp`
- failOverDisabledReturnsEarly
- failOverEnforcesMemberId
- failOverRejectsSmallLeaseTtl
- failOverValidAccepted
- checkAndGetValueRejectsNonNumeric

`NodeTimeMaintenanceTest.cpp`
- testNodeTimeMaintenance_doubleNode
- testNodeTimeMaintenance_singlarNode

</details>

<details><summary><b>bcos-utilities</b> — 8 files, 19 cases</summary>

`AnyHolderTest.cpp`
- inPlaceMoveAndConvert

`BoostLogInitializerTest.cpp`
- getLogLevelMapsAllSeverities
- initLogHourlyRotation
- initLogNonHourlyWithArchive
- initLogConsole
- initLogRejectsBadRotateTimePoint
- initLogRejectsTooSmallFileSize

`CommonUtilTest.cpp`
- timeAndDate
- intConversions
- stringChecks
- rateMath
- threadName

`ErrorHelperTest.cpp`
- buildErrorAndAccessors

`IOServicePoolTest.cpp`
- startRoundRobinStop

`RateCollectorTest.cpp`
- factoryBuildsCollector
- updateReportFlushCycle

`RateReporterTest.cpp`
- updateReportFlushCycle

`TokenBucketRateLimiterTest.cpp`
- overLimitNeverGranted
- settersRollbackAndStoredFastPath

</details>

### CODE REVIEW

**Review by ywy2090 — APPROVED** [review]

**Review by morebtcg — COMMENTED** [review]
## Coverage PR Review — #5228

Overall this is a well-structured coverage improvement PR. The tests deliberately target error paths and edge cases rather than just happy paths, which is the right approach for coverage. The discovery of 4 real bugs (#5353–#5356) validates the value of this effort.

Below are specific actionable suggestions on a few files.

File: bcos-boostssl/test/unittests/websocket/RawWsMessageTest.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    msg.setSeq("seq-1");
+    msg.setPacketType(7);
+
+    BOOST_REQUIRE(msg.payload());
+    BOOST_CHECK_EQUAL(msg.payload()->size(), 3U);
+    BOOST_CHECK_EQUAL(msg.seq(), "seq-1");
+    // packetType is fixed to WS_RAW_MESSAGE_TYPE for raw messages; just exercise
+    // the getter/setter without asserting a changed value.
+    BOOST_CHECK_NO_THROW(msg.packetType());
+
```
Comment (morebtcg) [inline bcos-boostssl/test/unittests/websocket/RawWsMessageTest.cpp:42]: **建议：对 `packetType()` 做更强的断言。**

当前测试只用了 `BOOST_CHECK_NO_THROW(msg.packetType())`，但注释里明确写了 "packetType is fixed to WS_RAW_MESSAGE_TYPE for raw messages"。既然值已知是固定的，应该直接断言返回值而非仅检查是否抛异常：

```cpp
BOOST_CHECK_EQUAL(msg.packetType(), WS_RAW_MESSAGE_TYPE);
```

同样 `setVersion`/`setExt` 是 no-op，getter 有明确定义的返回值的话也应该做值断言而非仅 `NO_THROW`。

File: bcos-boostssl/test/unittests/websocket/RawWsMessageTest.cpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    msg.setVersion(2);
+    msg.setExt(1);
+    BOOST_CHECK_NO_THROW(msg.version());
+    BOOST_CHECK_NO_THROW(msg.ext());
+}
+
+BOOST_AUTO_TEST_CASE(respPacketFlag)
+{
+    RawWsMessage msg;
+    msg.setRespPacket();
```
Comment (morebtcg) [inline bcos-boostssl/test/unittests/websocket/RawWsMessageTest.cpp:55]: **建议：`isRespPacket()` 应断言布尔返回值。**

`setRespPacket()` 设置了响应标志后，`isRespPacket()` 应返回 `true`。当前只用了 `BOOST_CHECK_NO_THROW(msg.isRespPacket())`，没有验证语义正确性。建议改为：

```cpp
msg.setRespPacket();
BOOST_CHECK(msg.isRespPacket());
```

File: bcos-tars-protocol/test/BlockImplTest.cpp [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+    block.inner().type = 3;  // mutable inner
+    BOOST_CHECK_EQUAL(static_cast<int>(block.blockType()), 3);
+    BOOST_CHECK_EQUAL(std::as_const(block).inner().type, 3);
+
+    bcostars::Block replacement;
+    replacement.type = 5;
+    block.setInner(std::move(replacement));
+    BOOST_CHECK_EQUAL(static_cast<int>(block.blockType()), 5);
+}
```
Comment (morebtcg) [inline bcos-tars-protocol/test/BlockImplTest.cpp:51]: **建议：补充越界访问的防御性测试。**

PR 描述中提到了 bug #5355（`BlockImpl::setReceipt`/`setTransaction`/`transactionHash` 使用未检查的索引导致 OOB）。当前测试 `transactionAndReceiptByIndex` 只覆盖了合法索引路径，建议增加越界索引的测试用例来验证边界行为：

```cpp
// 负数索引 / 超出 size 的索引
BOOST_CHECK_THROW(block.setReceipt(-1, receipt), std::out_of_range);
BOOST_CHECK_THROW(block.transactionHash(block.transactionsSize()), std::out_of_range);
```

这样可以在修复 #5355 后起到回归保护作用。

File: bcos-tool/test/unittests/libtool/NodeConfigLoadersTest.cpp [hunk 4]
```diff
@@ -0,0 +1,501 @@
+/*
```
Comment (morebtcg) [inline bcos-tool/test/unittests/libtool/NodeConfigLoadersTest.cpp]: **建议：考虑拆分为多个小文件以提高可维护性。**

这个测试文件有 501 行，覆盖了 `NodeConfig` 的十几个 `load*` 方法。随着更多 coverage 用例的加入会越来越臃肿。

建议按功能域拆分，例如：
- `NodeConfigLoadCertTest.cpp` — 证书相关
- `NodeConfigLoadConsensusTest.cpp` — 共识配置
- `NodeConfigLoadExecutorTest.cpp` — 执行器配置
- `NodeConfigLoadStorageTest.cpp` — 存储配置

这样每个文件职责单一，也更容易定位失败用例。

File: bcos-rpc/test/unittests/rpc/GatewayHandlersTest.cpp [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+template <typename Invoke>
+AsyncResult awaitHandler(Invoke&& invoke)
+{
+    auto shared = std::make_shared<std::promise<AsyncResult>>();
+    auto future = shared->get_future();
+    std::forward<Invoke>(invoke)([shared](bcos::Error::Ptr error, Json::Value&) {
+        shared->set_value(AsyncResult{true, error != nullptr});
+    });
+    return future.get();
```
Comment (morebtcg) [inline bcos-rpc/test/unittests/rpc/GatewayHandlersTest.cpp:52]: **小建议：`awaitHandler` 模板里 local `promise` 被 move 进了 `shared_ptr`，随后直接 discard。**

当前写法：
```cpp
auto shared = std::make_shared<std::promise<AsyncResult>>();
auto future = shared->get_future();
```

这里 `shared` 已经持有了 `promise`（通过 `make_shared` 原地构造），不需要额外的 local `promise` 变量。`RpcHandlersTest.cpp` 中的 `awaitHandler` 有相同模式但写法略有不同（先创建 local promise 再 move），建议统一风格。

File: bcos-tars-protocol/test/BlockHeaderImplTest.cpp [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+{
+    bcostars::BlockHeader tars;  // must outlive the wrapper below
+    tars.data.version = 3;
+    BlockHeaderImpl header(tars);
+    BOOST_CHECK_EQUAL(header.version(), 3U);
+
+    // mutable inner() writes back to the referenced struct
+    header.inner().data.version = 5;
+    BOOST_CHECK_EQUAL(header.version(), 5U);
+    BOOST_CHECK_EQUAL(std::as_const(header).inner().data.version, 5);
```
Comment (morebtcg) [inline bcos-tars-protocol/test/BlockHeaderImplTest.cpp:33]: **疑问：是否需要验证 `const` 版本的 `inner()` 返回的是引用而非副本？**

当前测试：
```cpp
BOOST_CHECK_EQUAL(std::as_const(header).inner().data.version, 5);
```

如果 `const inner()` 不小心返回了值类型（拷贝），这个断言仍会通过。建议额外验证修改外部 tars struct 后 const inner() 也反映变化，或使用 `&` 取地址比对：

```cpp
BOOST_CHECK_EQUAL(&std::as_const(header).inner(), &tars);
```

**Comment by kyonRay** [comment]
Thanks @morebtcg and Copilot — all review points addressed in 3dd2889f9. Summary of how each was resolved:

**Stronger assertions (was `BOOST_CHECK_NO_THROW`)**
- `RawWsMessageTest`: now asserts concrete values — `packetType() == WS_RAW_MESSAGE_TYPE`, `version()/ext() == 0`.
- `NodeConfigTest`: asserts `txpoolLimit() == 15000` / `notifyWorkerNum() == 2` after the partial load, instead of `BOOST_CHECK(true)`.
- `BlockHeaderImplTest`: asserts `&std::as_const(header).inner() == &tars` so a copy-returning `inner()` would fail.

**Misleading names/comments fixed**
- `SealingManager`: renamed to `resetSealingDisablesProposalGeneration`; comment corrected (`resetSealing` sets `sealingNumber = endSealingNumber + 1`, not `-1`).
- `JsonRpcRequest`: renamed the parse-false case to `...ReportsInvalidRequest` and added a distinct `...ReportsParseError` case for the field-type-error branch.

**Ineffective test fixed**
- `BcosKms`: now seeds the cache with a successful fetch, confirms the cache hit, then triggers a server error for a *different* cipher and asserts the previously-cached entry is re-fetched — so a broken `clearCache()` would now fail the test.

**Structural**
- Split the 501-line `NodeConfigLoadersTest.cpp` into `NodeConfigChainLoadersTest` / `NodeConfigServiceLoadersTest` / `NodeConfigInfraLoadersTest`, sharing a `NodeConfigLoaderProbe.h` (the `LoaderProbe` + `fromIni` helper).

**Three points where I diverged from the literal suggestion — please double-check:**

1. `RawWsMessage::isRespPacket()` — the suggested `BOOST_CHECK(msg.isRespPacket())` would fail: `setRespPacket()` is a no-op and `isRespPacket()` is hardcoded `false` for raw messages. The test now asserts `!isRespPacket()` (pinning the real behavior). If raw messages are meant to support the response flag, that's a production bug rather than a test issue.

2. `BlockImpl` OOB (bug #5355) — the suggested `BOOST_CHECK_THROW(setReceipt(-1, ...), std::out_of_range)` can't be added yet: the current code does an *unchecked* vector access (UB), so it crashes/corrupts rather than throwing. I added a comment marking this; the `std::out_of_range` regression test should land together with the bounds-check fix in #5355.

3. `awaitHandler` — `GatewayHandlersTest.cpp` already uses the clean in-place form; the redundant local-`promise` pattern was in `RpcHandlersTest.cpp`, so the simplification went there.

**Review by Shichen-Wu — APPROVED** [review]
